File: Web/src/domain/scores/tunes/naIvanaNaKupala.ts
import type { Score, ScoreEvent } from '../score.js'
import { blow, draw, rest } from '../scoreWriting.js'

const callingOverTheFire: ScoreEvent[] = [
    blow(6, 1), blow(6, 1), draw(5, 1),
    blow(5, 1), draw(4, 2, { vibrato: 0.5 }), blow(4, 1),
    draw(4, 1), blow(5, 1), draw(5, 1),
    blow(6, 2, { vibrato: 0.5 })
]

const theAnswerBack: ScoreEvent[] = [
    blow(6, 1), draw(6, 1), blow(6, 1),
    draw(5, 1), blow(5, 2, { vibrato: 0.5 }), draw(4, 1),
    blow(5, 1), draw(4, 1), blow(4, 1),
    blow(4, 2, { vibrato: 0.5 })
]

export const naIvanaNaKupala: Score = {
    name: 'Na Ivana na Kupala',
    key: 'G',
    position: 'first',
    beatsPerMinute: 104,
    events: [...callingOverTheFire, rest(1), ...theAnswerBack]
}

File: Apps/iOS/Domain/Entities/Scores/NaIvanaNaKupala.swift
extension Score {
    static let naIvanaNaKupala = Score(
        name: "Na Ivana na Kupala",
        key: .g,
        position: .first,
        beatsPerMinute: 104,
        events: callingOverTheFire + [.rest(beats: 1)] + theAnswerBack
    )

    private static let callingOverTheFire: [ScoreEvent] = [
        .blow(.six, 1), .blow(.six, 1), .draw(.five, 1),
        .blow(.five, 1), .draw(.four, 2, vibrato: 0.5), .blow(.four, 1),
        .draw(.four, 1), .blow(.five, 1), .draw(.five, 1),
        .blow(.six, 2, vibrato: 0.5)
    ]

    private static let theAnswerBack: [ScoreEvent] = [
        .blow(.six, 1), .draw(.six, 1), .blow(.six, 1),
        .draw(.five, 1), .blow(.five, 2, vibrato: 0.5), .draw(.four, 1),
        .blow(.five, 1), .draw(.four, 1), .blow(.four, 1),
        .blow(.four, 2, vibrato: 0.5)
    ]
}
